packages/alfa-rules/src/sia-r82/rule.ts

```diff
@@ -186,3 +119,62 @@ export namespace Outcomes {
     resolve it`
   );
 }
+
+function identifiesTarget(
+  indicators: Array<Node>,
+  error: Err<string>,
+  device: Device
+): Interview<Question, Node, Option.Maybe<Result<string, string>>> {
+  const indicator = indicators[0];
+
+  if (indicator === undefined) {
+    return error;
+  }
+
+  return Question.of(
+    "error-indicator-identifies-form-field",
+    "boolean",
+    indicator,
+    "Does the error indicator identify, in text, the form field it relates to?"
+  ).map((isIdentified) => {
+    if (isIdentified) {
+      if (test(isPerceivable(device), indicator)) {
+        return Outcomes.ErrorIndicatorIdentifiesTarget;
+      } else {
+        error = Outcomes.ErrorIndicatorIdentifiesTargetButIsNotPerceivable;
+      }
+    }
+
+    return identifiesTarget(indicators.slice(1), error, device);
+  });
+}
+
+function describesResolution(
```

Comment:
This feels some much like a `reduce` if it wasn't for the early escape…

Reply:
I agree 😄 It _could_ be implemented as a fold, but that seemed a little wasteful. I'll look into possibly encoding this "early return" fold as a common operation.

Reply:
Maybe a statefull function with `for (indicator of indicators)` and a `break` is enough…
I'm cringing so much at the recursive call to `f(array.slice(1))` which looks like we don't know how to iterate over elements in an array…

Reply:
The issue is that it's really constructing a deeply nested chain of `.map()` invocations, so neither a fold nor a loop will actually do us any good 😕